src/__tests__/incidents/view/ViewIncident.test.tsx

```diff
@@ -12,69 +14,291 @@ import ViewIncidentDetails from '../../../incidents/view/ViewIncidentDetails'
 import * as titleUtil from '../../../page-header/title/TitleContext'
+import TextFieldWithLabelFormGroup from '../../../shared/components/input/TextFieldWithLabelFormGroup'
 import IncidentRepository from '../../../shared/db/IncidentRepository'
 import Incident from '../../../shared/model/Incident'
+import Note from '../../../shared/model/Note'
 import Permissions from '../../../shared/model/Permissions'
+import NewNoteModal from '../../../shared/notes/NewNoteModal'
 import { RootState } from '../../../shared/store'
+import * as uuid from '../../../shared/util/uuid'
 
 const { TitleProvider } = titleUtil
 const mockStore = createMockStore<RootState, any>([thunk])
 
 describe('View Incident', () => {
-  const setup = async (permissions: Permissions[]) => {
+  let incidentRepositorySaveSpy: any
+  const queryCache = new QueryCache()
+  const expectedResolveDate = new Date()
+  const mockedIncident: Incident = {
+    id: '1234',
+    code: 'some code',
+    department: 'some department',
+    description: 'some description',
+    category: 'some category',
+    categoryItem: 'some category item',
+    status: 'reported',
+    reportedBy: 'some user id',
+    reportedOn: new Date().toISOString(),
+    date: new Date().toISOString(),
+    notes: [
+      {
+        id: '2345',
+        date: 'some date',
+        text: 'some note text',
+        givenBy: 'given by text',
+      } as Note,
+    ],
+  } as Incident
+
+  const setup = async (permissions: Permissions[], incident: Incident) => {
     jest.resetAllMocks()
+    Date.now = jest.fn(() => expectedResolveDate.valueOf())
+    jest.spyOn(uuid, 'uuid').mockReturnValue('7777')
     jest.spyOn(titleUtil, 'useUpdateTitle').mockImplementation(() => jest.fn())
     jest.spyOn(breadcrumbUtil, 'default')
-    jest.spyOn(IncidentRepository, 'find').mockResolvedValue({
-      id: '1234',
-      date: new Date().toISOString(),
-      code: 'some code',
-      reportedOn: new Date().toISOString(),
-    } as Incident)
+    jest.spyOn(IncidentRepository, 'find').mockResolvedValue(incident)
     const history = createMemoryHistory()
     history.push(`/incidents/1234`)
+    incidentRepositorySaveSpy = jest
+      .spyOn(IncidentRepository, 'saveOrUpdate')
+      .mockResolvedValue(mockedIncident)
 
     const store = mockStore({
       user: {
         permissions,
+        user: {
+          id: '8542',
+        },
       },
     } as any)
 
     let wrapper: any
     await act(async () => {
       wrapper = await mount(
-        <ButtonBarProvider.ButtonBarProvider>
-          <Provider store={store}>
-            <Router history={history}>
-              <Route path="/incidents/:id">
-                <TitleProvider>
-                  <ViewIncident />
-                </TitleProvider>
-              </Route>
-            </Router>
-          </Provider>
-        </ButtonBarProvider.ButtonBarProvider>,
+        <ReactQueryCacheProvider queryCache={queryCache}>
```

Comment:
Could be wrong here, but I don't think a query cache needs to be provided unless a React Query hook is being explicitly called and tested (as opposed to being mocked).